src/LayerEditor/ui/dialogs/regions/add_region.py

```diff
@@ -76,9 +77,15 @@ def __init__(self, parent=None):
         self.region_dim.addItem(self.REGION_DESCRIPTION[RegionDim.well], RegionDim.well)
         self.region_dim.addItem(self.REGION_DESCRIPTION[RegionDim.fracture], RegionDim.fracture)
         self.region_dim.addItem(self.REGION_DESCRIPTION[RegionDim.bulk], RegionDim.bulk)
-        self.region_dim.setCurrentIndex(3) 
-        grid.addWidget(d_region_dim , 1, 0)
-        grid.addWidget(self.region_dim , 1, 1)
+        self.region_dim.setCurrentIndex(on_creation_dim)
+
+        def adjust_name(idx):
+            if self.region_name.text()[0:5] in ["(0D) ", "(1D) ", "(2D) ", "(3D) "]:
```

Comment:
- use QLineEdit::textEdited slot to set an indicator for a changed region name
- use true region dimension for the default region name, that is dimension selected from the d_region_dim widget + 0/1  for fracture layer/bulk layer respectively
- do not use parenthesis and spaces in the region name, only troubles through the further processing